tests/test_facebook_archived_data.py

```diff
@@ -0,0 +1,112 @@
+        ##########################################################################
+
+        # create a new connection with the include_deleted = true
+        conn_id_2 = connections.ensure_connection(self, original_properties=False)
+
+        # run check mode
+        found_catalogs_2 = self.run_and_verify_check_mode(conn_id_2)
+
+        # table and field selection
+        test_catalogs_2_all_fields = [catalog for catalog in found_catalogs_2
+                                      if catalog.get('tap_stream_id') in expected_streams]
+        self.perform_and_verify_table_and_field_selection(conn_id_2, test_catalogs_2_all_fields, select_all_fields=True)
+
+        # run sync
+        record_count_by_stream_2 = self.run_and_verify_sync(conn_id_2)
+        synced_records_2 = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # expected primary keys
+                expected_primary_keys = self.expected_primary_keys()[stream]
+
+                # collect information about count of records
+                record_count_sync_1 = record_count_by_stream_1.get(stream, 0)
+                record_count_sync_2 = record_count_by_stream_2.get(stream, 0)
+
+                # collect list and set of primary keys for all the records
+                primary_keys_list_1 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                                       for message in synced_records_1.get(stream).get('messages')
+                                       if message.get('action') == 'upsert']
+                primary_keys_list_2 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                                       for message in synced_records_2.get(stream).get('messages')
+                                       if message.get('action') == 'upsert']
+                primary_keys_sync_1 = set(primary_keys_list_1)
+                primary_keys_sync_2 = set(primary_keys_list_2)
+
+                # collect list of effective_status for all the records
+                records_status_sync1 = [message.get('data').get('effective_status')
+                                       for message in synced_records_1.get(stream).get('messages')
+                                       if message.get('action') == 'upsert']
+                records_status_sync2 = [message.get('data').get('effective_status')
+                                       for message in synced_records_2.get(stream).get('messages')
+                                       if message.get('action') == 'upsert']
+
+                # Verifying that no ARCHIVED records are returned for sync 1
+                self.assertTrue('ARCHIVED' not in records_status_sync1)
+
+                # Verifying that ARCHIVED records are returned for sync 2
+                self.assertTrue('ARCHIVED' in records_status_sync2)
```

Comment:
```suggestion
                self.assertNotIn('ARCHIVED', records_status_sync1)

                # Verifying that ARCHIVED records are returned for sync 2
                self.assertIn('ARCHIVED', records_status_sync2)
```

Reply:
I find that `assertTrue` throws a harder-to-debug AssertionError. Something like `False != True` as opposed to `'ARCHIVED not in records_status_sync2` when using `assertIn`.